HTTP_API.md

```diff
@@ -0,0 +1,952 @@
+    {"downloadUrl":"http://0.0.0.0:8181/api/v1/targets/service:jmx:rmi:%2F%2F%2Fjndi%2Frmi:%2F%2Flocalhost:9091%2Fjmxrmi/recordings/foo","reportUrl":"http://0.0.0.0:8181/api/v1/targets/service:jmx:rmi:%2F%2F%2Fjndi%2Frmi:%2F%2Flocalhost:9091%2Fjmxrmi/reports/foo","id":1,"name":"foo","state":"RUNNING","startTime":1599847667483,"duration":5000,"continuous":false,"toDisk":true,"maxSize":0,"maxAge":0}
+    ```
+
+
+* #### `TargetRecordingUploadPostHandler`
+
+    ###### synopsis
+    Uploads a recording of a target JVM to the Grafana datasource
+    that Container JFR is configured with
+    (determined by the environment variable `GRAFANA_DATASOURCE_URL`).
+
+    ###### request
+    `POST /api/v1/targets/:targetId/recordings/:recordingName/upload`
+
+    `targetId` - The location of the target JVM to connect to,
+    in the form of a `service:rmi:jmx://` JMX Service URL, or `hostname:port`.
+
+    `recordingName` - The name of the recording to upload.
+
+    ###### response
+    `200` - The body is the body from the response that Container JFR got
+    after sending the upload request to the Grafana datasource.
+
+    `401` - User authentication failed. The body is an error message.
+
+    `404` - The target or the recording could not be found.
+    The body is an error message.
+
+    `407` - JMX authentication failed. The body is an error message.
+
+    `500` - There was an unexpected error. The body is an error message.
+
+    `501` - The Grafana datasource URL is malformed.
+    The body is an error message.
+
+    `502` - Container JFR received an invalid response from the
+    Grafana datasource after sending the upload request.
+    The body is an error message.
+
+    ###### example
+    ```
+    $ curl -X POST localhost:8181/api/v1/targets/localhost/recordings/foo/upload
+    Uploaded: file-uploads/72ee43b8-c858-4aef-ae7e-4bcf7d93ec7c
+    Loaded: file-uploads/72ee43b8-c858-4aef-ae7e-4bcf7d93ec7c
+    ```
+
+
+* #### `TargetReportGetHandler`
+
+    ###### synopsis
+    Returns a report of a recording of a target JVM.
+
+    ###### request
+    `GET /api/v1/targets/:targetId/reports/:recordingName`
+
+    `targetId` - The location of the target JVM to connect to,
+    in the form of a `service:rmi:jmx://` JMX Service URL, or `hostname:port`.
+
+    `recordingName` - The name of the recording to get the report for.
+
+    ###### response
+    `200` - The body is the requested report.
+
+    `401` - User authentication failed. The body is an error message.
+
+    `404` - The report could not be found, or the target could not be found.
+    The body is an error message.
+
+    `407` - JMX authentication failed. The body is an error message.
+
+    `500` - There was an unexpected error. The body is an error message.
+
+    ###### example
+    ```
+    [vma@victor-work ~]$ curl localhost:8181/api/v1/targets/localhost/reports/foo --output report.html
```

Comment:
I think the shell prompt stuff here `[vma@victor-work ~]` should be removed for consistency with the rest of the document

Reply:
Whoops, yeah that should've been removed. I actually found another one hidden at the end of the example for `TargetTemplatesGetHandler`!